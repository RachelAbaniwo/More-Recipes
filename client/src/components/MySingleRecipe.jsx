import React from 'react';
import { Link } from 'react-router';
import PropTypes from 'prop-types';

/**
 * Displays a single recipe
 * @function MySingleRecipe
 *
 * @param {object} props
 *
 * @returns {jsx} jsx for single recipe
 */
const MySingleRecipe = ({ recipe }) => ((
  <section className="col-sm-3 card-1 my-single" >
    <figure className="figure">
      <Link to={`view-recipe/${recipe.id}`}>
        <img
          src={recipe.recipeImage}
          alt="Search Result"
          className="figure-img img-thumbnail mx-auto single-recipe"
        />
      </Link>
      <figcaption className="figure-caption" id="my-single" >
        <Link
          to={`view-recipe/${recipe.id}`}
          className="single-recipe-name"
        >{recipe.name}
        </Link>
        <p>
          <i
            className="fa fa-thumbs-up text-info"
            aria-hidden="true"
          />{recipe.upvotes} &nbsp;
          <i
            className="fa fa-thumbs-down text-danger"
            aria-hidden="true"
          />{recipe.downvotes} &nbsp;
          <i
            className="fa fa-heart"
            aria-hidden="true"
            style={{ color: 'orange' }}
          />{recipe.favorites}
        </p>
      </figcaption>
    </figure>
  </section>
));

MySingleRecipe.propTypes = {
  recipe: PropTypes.shape({
    id: PropTypes.number.isRequired,
    upvotes: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    recipeImage: PropTypes.string
  }).isRequired
};


export default MySingleRecipe;
