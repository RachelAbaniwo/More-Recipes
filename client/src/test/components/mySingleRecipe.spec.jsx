import React from 'react';
import { shallow } from 'enzyme';
import recipeMock from '../mockData/recipeMock';
import MySingleRecipe from '../../components/MySingleRecipe';


describe('The MySingleRecipe component', () => {
  it('should mount without crashing', () => {
    const wrapper = shallow(<MySingleRecipe recipe={recipeMock} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('img').length).toBe(1);
  });
});
