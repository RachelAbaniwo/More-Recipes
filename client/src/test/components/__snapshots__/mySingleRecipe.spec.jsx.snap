// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MySingleRecipe component should mount without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MySingleRecipe
    recipe={
      Object {
        "User": Object {
          "id": 1,
          "username": "test_user",
        },
        "category": "breakfast",
        "description": "Fried ofada rice",
        "id": 1,
        "method": "fry",
        "name": "Chicken",
        "recipeImage": "http://test.recipe_image",
        "upvotes": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <figure
        className="figure"
      >
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="view-recipe/1"
        >
          <img
            alt="Search Result"
            className="figure-img img-thumbnail mx-auto single-recipe"
            src="http://test.recipe_image"
          />
        </Link>
        <figcaption
          className="figure-caption"
          id="my-single"
        >
          <Link
            className="single-recipe-name"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="view-recipe/1"
          >
            Chicken
          </Link>
          <p>
            <i
              aria-hidden="true"
              className="fa fa-thumbs-up text-info"
            />
            1
              
            <i
              aria-hidden="true"
              className="fa fa-thumbs-down text-danger"
            />
              
            <i
              aria-hidden="true"
              className="fa fa-heart"
              style={
                Object {
                  "color": "orange",
                }
              }
            />
          </p>
        </figcaption>
      </figure>,
      "className": "col-sm-3 card-1 my-single",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="view-recipe/1"
          >
            <img
              alt="Search Result"
              className="figure-img img-thumbnail mx-auto single-recipe"
              src="http://test.recipe_image"
            />
          </Link>,
          <figcaption
            className="figure-caption"
            id="my-single"
          >
            <Link
              className="single-recipe-name"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="view-recipe/1"
            >
              Chicken
            </Link>
            <p>
              <i
                aria-hidden="true"
                className="fa fa-thumbs-up text-info"
              />
              1
                
              <i
                aria-hidden="true"
                className="fa fa-thumbs-down text-danger"
              />
                
              <i
                aria-hidden="true"
                className="fa fa-heart"
                style={
                  Object {
                    "color": "orange",
                  }
                }
              />
            </p>
          </figcaption>,
        ],
        "className": "figure",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              alt="Search Result"
              className="figure-img img-thumbnail mx-auto single-recipe"
              src="http://test.recipe_image"
            />,
            "onlyActiveOnIndex": false,
            "style": Object {},
            "to": "view-recipe/1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Search Result",
              "className": "figure-img img-thumbnail mx-auto single-recipe",
              "src": "http://test.recipe_image",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="single-recipe-name"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="view-recipe/1"
              >
                Chicken
              </Link>,
              <p>
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-up text-info"
                />
                1
                  
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-down text-danger"
                />
                  
                <i
                  aria-hidden="true"
                  className="fa fa-heart"
                  style={
                    Object {
                      "color": "orange",
                    }
                  }
                />
              </p>,
            ],
            "className": "figure-caption",
            "id": "my-single",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Chicken",
                "className": "single-recipe-name",
                "onlyActiveOnIndex": false,
                "style": Object {},
                "to": "view-recipe/1",
              },
              "ref": null,
              "rendered": "Chicken",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    aria-hidden="true"
                    className="fa fa-thumbs-up text-info"
                  />,
                  1,
                  "  ",
                  <i
                    aria-hidden="true"
                    className="fa fa-thumbs-down text-danger"
                  />,
                  undefined,
                  "  ",
                  <i
                    aria-hidden="true"
                    className="fa fa-heart"
                    style={
                      Object {
                        "color": "orange",
                      }
                    }
                  />,
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-thumbs-up text-info",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                1,
                "  ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-thumbs-down text-danger",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                undefined,
                "  ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-heart",
                    "style": Object {
                      "color": "orange",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                undefined,
              ],
              "type": "p",
            },
          ],
          "type": "figcaption",
        },
      ],
      "type": "figure",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <figure
          className="figure"
        >
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="view-recipe/1"
          >
            <img
              alt="Search Result"
              className="figure-img img-thumbnail mx-auto single-recipe"
              src="http://test.recipe_image"
            />
          </Link>
          <figcaption
            className="figure-caption"
            id="my-single"
          >
            <Link
              className="single-recipe-name"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="view-recipe/1"
            >
              Chicken
            </Link>
            <p>
              <i
                aria-hidden="true"
                className="fa fa-thumbs-up text-info"
              />
              1
                
              <i
                aria-hidden="true"
                className="fa fa-thumbs-down text-danger"
              />
                
              <i
                aria-hidden="true"
                className="fa fa-heart"
                style={
                  Object {
                    "color": "orange",
                  }
                }
              />
            </p>
          </figcaption>
        </figure>,
        "className": "col-sm-3 card-1 my-single",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              onlyActiveOnIndex={false}
              style={Object {}}
              to="view-recipe/1"
            >
              <img
                alt="Search Result"
                className="figure-img img-thumbnail mx-auto single-recipe"
                src="http://test.recipe_image"
              />
            </Link>,
            <figcaption
              className="figure-caption"
              id="my-single"
            >
              <Link
                className="single-recipe-name"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="view-recipe/1"
              >
                Chicken
              </Link>
              <p>
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-up text-info"
                />
                1
                  
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-down text-danger"
                />
                  
                <i
                  aria-hidden="true"
                  className="fa fa-heart"
                  style={
                    Object {
                      "color": "orange",
                    }
                  }
                />
              </p>
            </figcaption>,
          ],
          "className": "figure",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="Search Result"
                className="figure-img img-thumbnail mx-auto single-recipe"
                src="http://test.recipe_image"
              />,
              "onlyActiveOnIndex": false,
              "style": Object {},
              "to": "view-recipe/1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Search Result",
                "className": "figure-img img-thumbnail mx-auto single-recipe",
                "src": "http://test.recipe_image",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  className="single-recipe-name"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="view-recipe/1"
                >
                  Chicken
                </Link>,
                <p>
                  <i
                    aria-hidden="true"
                    className="fa fa-thumbs-up text-info"
                  />
                  1
                    
                  <i
                    aria-hidden="true"
                    className="fa fa-thumbs-down text-danger"
                  />
                    
                  <i
                    aria-hidden="true"
                    className="fa fa-heart"
                    style={
                      Object {
                        "color": "orange",
                      }
                    }
                  />
                </p>,
              ],
              "className": "figure-caption",
              "id": "my-single",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Chicken",
                  "className": "single-recipe-name",
                  "onlyActiveOnIndex": false,
                  "style": Object {},
                  "to": "view-recipe/1",
                },
                "ref": null,
                "rendered": "Chicken",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      aria-hidden="true"
                      className="fa fa-thumbs-up text-info"
                    />,
                    1,
                    "  ",
                    <i
                      aria-hidden="true"
                      className="fa fa-thumbs-down text-danger"
                    />,
                    undefined,
                    "  ",
                    <i
                      aria-hidden="true"
                      className="fa fa-heart"
                      style={
                        Object {
                          "color": "orange",
                        }
                      }
                    />,
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "className": "fa fa-thumbs-up text-info",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  1,
                  "  ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "className": "fa fa-thumbs-down text-danger",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  undefined,
                  "  ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "className": "fa fa-heart",
                      "style": Object {
                        "color": "orange",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  undefined,
                ],
                "type": "p",
              },
            ],
            "type": "figcaption",
          },
        ],
        "type": "figure",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
